lebedigital/demonstrator_scripts/computation_volume_content.py

```diff
@@ -44,66 +46,93 @@ def computation_volume_content(input_dic):
     output = {}
 
     # optional parameter
-    if 'plasticizer_volume_content' not in input_dic.keys():
-        input_dic['plasticizer_volume_content'] = 0 * ureg('kg/m^3')
-        input_dic['density_plasticizer'] = 42 * ureg('kg/m^3')  # dummy value
+    if "plasticizer_volume_content" not in input_dic.keys():
+        input_dic["plasticizer_volume_content"] = 0 * ureg("kg/m^3")
+        input_dic["density_plasticizer"] = 42 * ureg("kg/m^3")  # dummy value
 
     # converting to correct pint units / automatic check for pint input / check for required input values
-    input_dic['density_sub'].ito('kg/m^3')
-    input_dic['density_cem'].ito('kg/m^3')
-    input_dic['density_water'].ito('kg/m^3')
-    input_dic['density_plasticizer'].ito('kg/m^3')
-    input_dic['density_aggregates'].ito('kg/m^3')
+    input_dic["density_sub"].ito("kg/m^3")
+    input_dic["density_cem"].ito("kg/m^3")
+    input_dic["density_water"].ito("kg/m^3")
+    input_dic["density_plasticizer"].ito("kg/m^3")
+    input_dic["density_aggregates"].ito("kg/m^3")
+
+    # compute volume fraction of substitute to cement
+    if input_dic["sc_mass_fraction"] == 0:
+        output["sc_volume_fraction"] = 0
+    else:
+        mass_sub = 1 * ureg("kg") / (1 + (1 - input_dic["sc_mass_fraction"]) / input_dic["sc_mass_fraction"])
```

Comment:
I will double check.
I think I will just integrate the "computation_sc_volume_fraction" into this script (it might already be integrated, this implementation was a bit rushed...

Reply:
Fixed it, this implementation works but is unnecessarily complex.